### ISSUE TITLE
error: timed out waiting for the condition on jobs/istio-init-crd-10-1.3.0

### ISSUE THREAD

**Issue by adurai81** [istio/istio#21317]
**Bug description**
Intermittently we see istio installation fails due to CRDs not getting initialized in time.

**Expected behavior**
Istio installation should go through without any issues.

**Steps to reproduce the bug**
See istio installation section below.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

kubectl version
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.7", GitCommit:"4f504dd9ee54a3621502518bc64f0df487587d12", GitTreeState:"clean", BuildDate:"2020-01-23T00:34:28Z", GoVersion:"go1.12.7 BoringCrypto", Compiler:"gc", Platform:"linux/amd64"}
```

Helm Version
```
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14+unreleased", GitCommit:"", GitTreeState:"clean"}
```

**How was Istio installed?**
```
# Variables
VERSION=1.3.0

# Istio namespace
kubectl  --kubeconfig=${_kubeconfig_}  create namespace istio-system

# Install CRDs
helm template $(dirname $0)/istio-init --name istio-init --namespace istio-system | kubectl  --kubeconfig=${_kubeconfig_} apply -f -

# Verify if the CRDs are installed
kubectl  --kubeconfig=${_kubeconfig_} -n istio-system wait --for=condition=complete job/istio-init-crd-10-$VERSION
kubectl  --kubeconfig=${_kubeconfig_} -n istio-system wait --for=condition=complete job/istio-init-crd-11-$VERSION
kubectl  --kubeconfig=${_kubeconfig_} -n istio-system wait --for=condition=complete job/istio-init-crd-12-$VERSION

# Install Istio
helm template $(dirname $0)/istio --name istio --namespace istio-system | kubectl  --kubeconfig=${_kubeconfig_}  apply -f -
```

**Environment where bug was observed (cloud vendor, OS, etc)**
Oracle Cloud Infrastructure

**Comment by dgn** [comment]
Hi @adurai81, is this potentially due to a longer delay in applying these manifests? Try setting a longer timeout (e.g. `--timeout 5m`) on the `kubectl wait` commands

**Comment by adurai81** [comment]
@dgn I used to follow the `wait` approach, but then I don't know (can't estimate) how much to wait. 

As suggested by @howardjohn on the forum [1], I followed the approach of relying on the job to ensure the CRDs are installed and available.

```
[1] - https://discuss.istio.io/t/automating-istio-installation-best-practice/3868
```

To your first question I don't know what the cause of the issue. I don't have the cluster any longer. Waiting for repro scenario. Will post here if I can get to the issue again.
